resources/user.go

```diff
@@ -0,0 +1,305 @@
+// Mgmt
+// Copyright (C) 2013-2017+ James Shubin and the project contributors
+// Written by James Shubin <james@shubin.ca> and the project contributors
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package resources
+
+import (
+	"fmt"
+	"log"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"syscall"
+
+	"github.com/purpleidea/mgmt/recwatch"
+
+	errwrap "github.com/pkg/errors"
+)
+
+func init() {
+	RegisterResource("user", func() Res { return &UserRes{} })
+}
+
+const passwdFile = "/etc/passwd"
+
+// UserRes is a user account resource.
+type UserRes struct {
+	BaseRes           `yaml:",inline"`
+	State             string  `yaml:"state"` // state: exists, absent
+	AllowDuplicateUID bool    `yaml:"allowduplicateuid"`
```

Comment:
Nitpick, but I'd recommend putting the option modifier last in the list, sorry. More logical in that it's a thing that's less commonly used. (Same order in Compare)